test/integration/preemption_toleration_test.go

```diff
@@ -0,0 +1,310 @@
+			// prepare test cluster
+			registry := fwkruntime.Registry{preemptiontoleration.Name: preemptiontoleration.New}
+			cfg, err := util.NewDefaultSchedulerComponentConfig()
+			if err != nil {
+				t.Fatal(err)
+			}
+			cfg.Profiles[0].Plugins.PostFilter = schedapi.PluginSet{
+				Enabled: []schedapi.Plugin{
+					{Name: preemptiontoleration.Name},
+				},
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			}
+			cfg.Profiles[0].PluginConfig = append(cfg.Profiles[0].PluginConfig, schedapi.PluginConfig{
+				Name: preemptiontoleration.Name,
+				Args: &schedconfig.PreemptionTolerationArgs{
+					MinCandidateNodesPercentage: 10,
+					MinCandidateNodesAbsolute:   100,
+				},
+			})
+			testCtx := util.InitTestSchedulerWithOptions(
+				t,
+				testutils.InitTestAPIServer(t, "sched-preemptiontoleration", nil),
+				true,
+				scheduler.WithProfiles(cfg.Profiles[0]),
+				scheduler.WithFrameworkOutOfTreeRegistry(registry),
+				scheduler.WithPodInitialBackoffSeconds(int64(0)),
+				scheduler.WithPodMaxBackoffSeconds(int64(0)),
+			)
+			defer testutils.CleanupTest(t, testCtx)
+
+			cs, ns := testCtx.ClientSet, testCtx.NS.Name
+
+			// Create test node
+			if _, err := cs.CoreV1().Nodes().Create(testCtx.Ctx, node, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create node: %v", err)
+			}
+
+			// create test priorityclass for evaluating PreemptionToleration policy
+			if _, err := cs.SchedulingV1().PriorityClasses().Create(testCtx.Ctx, tt.priorityClass, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create priorityclass %s: %v", tt.priorityClass.Name, err)
+			}
+
+			// Create victim candidate pod and it should be scheduled
+			if _, err := cs.CoreV1().Pods(ns).Create(testCtx.Ctx, victimCandidate, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("failed to create victim candidate pod %q: %v", victimCandidate.Name, err)
+			}
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+				return podScheduled(cs, ns, victimCandidate.Name), nil
+			}); err != nil {
+				t.Fatalf("victim candidate pod %q failed to be scheduled: %v", victimCandidate.Name, err)
+			}
+			// simulate pod scheduled time
+			if err := wait.Poll(1*time.Second, 60*time.Second, func() (bool, error) {
+				vc, err := cs.CoreV1().Pods(ns).Get(testCtx.Ctx, victimCandidate.Name, metav1.GetOptions{})
+				if err != nil {
+					return false, err
+				}
+				vc.Status.Conditions = []v1.PodCondition{{
+					Type:               v1.PodScheduled,
+					Status:             v1.ConditionTrue,
+					LastTransitionTime: metav1.Time{Time: time.Now()},
+				}}
+				_, err = cs.CoreV1().Pods(ns).UpdateStatus(testCtx.Ctx, vc, metav1.UpdateOptions{})
+				if err != nil {
+					return false, err
+				}
+				return true, nil
+			}); err != nil {
+				t.Fatalf("failed to update victim candidate pod %q scheduled time: %v", victimCandidate.Name, err)
+			}
+
+			// Create the preemptor pod.
+			<-time.After(tt.waitPreemptorCreation)
```

Comment:
What's the intention here?

Reply:
This delay is intended to make sure that preemption toleration seconds on the victim's priority class is elapsed.

Reply:
Do it make sense to shorten the wait time by setting the `LastTransitionTime` as "time.Now() - wait time"? so we don't need to wait for `Sum(waitPreemptorCreation)` of all sub-tests.

Reply:
BTW: how long does it take in your local testing to run all sub-tests?

Reply:
> Do it make sense to shorten the wait time by setting the LastTransitionTime as "time.Now() - wait time"?

Thanks! This makes sense!  I removed the wait in https://github.com/kubernetes-sigs/scheduler-plugins/pull/224/commits/aec5378e5393929b2d2d16d51384bcb7bf072c6a.

And, I renamed `waitPreemptorCreation` with `simulateVictimScheduledBefore`.

> BTW: how long does it take in your local testing to run all sub-tests?

Before removing the wait, it was 180s (see https://github.com/kubernetes-sigs/scheduler-plugins/pull/224/#pullrequestreview-819905689).  After removing the wait, it became 103s!

```shell
$ git rev-parse HEAD
aec5378e5393929b2d2d16d51384bcb7bf072c6a
$ go test -timeout 30m -run ^TestPreemptionTolerationPlugin$ sigs.k8s.io/scheduler-plugins/test/integration
ok      sigs.k8s.io/scheduler-plugins/test/integration  103.527s
```